sdks/java/testing/load-tests/src/main/java/org/apache/beam/sdk/loadtests/SideInputLoadTest.java

```diff
@@ -0,0 +1,243 @@
+        break;
+      case LIST:
+        performTestWithList(input, syntheticStep);
+        break;
+    }
+  }
+
+  private void performTestWithList(
+      PCollection<KV<byte[], byte[]>> input, Optional<SyntheticStep> syntheticStep) {
+    applyStepIfPresent(input, "Synthetic step", syntheticStep);
+    PCollectionView<List<KV<byte[], byte[]>>> sideInput = input.apply(View.asList());
+    input
+        .apply(ParDo.of(new SideInputTestWithList(sideInput)).withSideInputs(sideInput))
+        .apply("Collect end time metrics", ParDo.of(runtimeMonitor));
+  }
+
+  private void performTestWithHashMap(
+      PCollection<KV<byte[], byte[]>> input, Optional<SyntheticStep> syntheticStep) {
+    applyStepIfPresent(input, "Synthetic step", syntheticStep);
+    PCollectionView<Map<byte[], byte[]>> sideInput = input.apply(View.asMap());
+    input
+        .apply(ParDo.of(new SideInputTestWithHashMap(sideInput)).withSideInputs(sideInput))
+        .apply("Collect end time metrics", ParDo.of(runtimeMonitor));
+  }
+
+  private void performTestWithIterable(
+      PCollection<KV<byte[], byte[]>> input, Optional<SyntheticStep> syntheticStep) {
+    applyStepIfPresent(input, "Synthetic step", syntheticStep);
+    PCollectionView<Iterable<KV<byte[], byte[]>>> sideInput;
+    if (options.getWindowCount() != 0) {
+      long windowDurationMilis = sourceOptions.numRecords / options.getWindowCount();
+      sideInput =
+          input
+              .apply(Window.into(FixedWindows.of(Duration.millis(windowDurationMilis))))
+              .apply(View.asIterable());
+    } else {
+      sideInput = input.apply(View.asIterable());
+    }
+
+    input
+        .apply(ParDo.of(new SideInputTestWithIterable(sideInput)).withSideInputs(sideInput))
+        .apply("Collect end time metrics", ParDo.of(runtimeMonitor));
+  }
+
+  private static class AddTimestamps extends DoFn<KV<byte[], byte[]>, KV<byte[], byte[]>> {
+    private static Instant timestamp = TIME;
+
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+      timestamp = TIME.plus(1L);
+      c.outputWithTimestamp(c.element(), timestamp);
+    }
+  }
+
+  private static class SideInputTestWithList extends DoFn<KV<byte[], byte[]>, KV<byte[], byte[]>> {
+
+    private final PCollectionView<List<KV<byte[], byte[]>>> sideInput;
+
+    public SideInputTestWithList(PCollectionView<List<KV<byte[], byte[]>>> sideInput) {
+      this.sideInput = sideInput;
+    }
+
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+      List<KV<byte[], byte[]>> si = c.sideInput(sideInput);
+
+      for (KV<byte[], byte[]> sideInputElement : si) {
+        // for every _input_ element iterate over all _sideInput_ elements
+        // count consumed bytes, examine memory usage, etc (Metrics API).
+        byte[] key = sideInputElement.getKey();
+      }
+    }
+  }
+
+  private class SideInputTestWithHashMap extends DoFn<KV<byte[], byte[]>, KV<byte[], byte[]>> {
```

Comment:
This tests `Map`, not `HashMap`. The runner (or SDK harness in portability) can choose whatever kind of `Map` it wants.

Reply:
Right, I'll rename it